### PR TITLE
Implement `NullStringConstraint`

### PR DESCRIPTION
**Pull request by libo**
Useful to validate against nil, '', 'undefined'

cc: @staugaard @morten

### CODE REVIEW

**Comment by staugaard** [comment]
It seams like multiple people find this useful. Feel free to merge. :+1:

I do find it odd thought that `null_string` does not match the string `"null"` though.

Maybe it should, or maybe it should be renamed to `undefined_string` ? Or maybe `nil_string` ?

**Comment by libo** [comment]
Wow blast from the past! 
Totally forgot about it... :-)

Yes I agree about probably a different naming it's a good idea.
Can you cc some of the people who are willing for this code to be merged and ask them too.
I will rebase and fix test...

**Comment by ciaranarcher** [comment]
I'll get on it tomorrow.

**Comment by ciaranarcher** [comment]
I was thinking we should only allow `""` (empty string). Then we can call it `EmptyStringConstraint`.

But before we do that I wanted to know why is `"undefined"` is there. Is that to deal with JS clients? If so, should we not reject that as being  a problem that belongs to the client?

Thoughts?

File: lib/stronger_parameters/constraints/null_string_contraint.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  class NullStringConstraint < Constraint
+    NULL_VALUES = ['', 'undefined'].freeze
+
+    def value(v)
+      if v.nil?
```
Comment (grosser) [inline lib/stronger_parameters/constraints/null_string_contraint.rb]: should not accept nil

Comment (grosser) [inline lib/stronger_parameters/constraints/null_string_contraint.rb]: <-> "NullString" sounds like a String to me, maybe `JavascriptNull` or something like that ?

**Comment by grosser** [comment]
cleaned it up and changed to NilString good to go ?
@libo @ciaranarcher @staugaard @kintner

File: test/null_string_constraints_test.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'test_helper'
```
Comment (pschambacher) [inline test/null_string_constraints_test.rb]: Rename file ```nil_string_constraints_test.rb``` ?

File: lib/stronger_parameters/constraints/nil_string_contraint.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  class NilStringConstraint < Constraint
+    NULL_VALUES = [nil, '', 'undefined', 'UNDEFINED']
```
Comment (pschambacher) [inline lib/stronger_parameters/constraints/nil_string_contraint.rb]: What about ```Undefined```? :trollface: 
Maybe we should check ```/undefined/i.match(v)``` ?

**Comment by pschambacher** [comment]
:+1:

File: lib/stronger_parameters/constraints/nil_string_contraint.rb (hunk 3 above)
Comment (grosser) [inline lib/stronger_parameters/constraints/nil_string_contraint.rb]: I think the only value that reasonable to happen is 'undefined' from
someone screwing up their js ... I'll check the logs for other usages ...

**Comment by kintner** [comment]
:+1:

**Comment by libo** [comment]
:+1: Thank you @grosser !

File: lib/stronger_parameters/constraints/nil_string_contraint.rb (hunk 3 above)
Comment (grosser) [inline lib/stronger_parameters/constraints/nil_string_contraint.rb]: found a bunch of 'undefined' but no 'UNDEFINED' so should be good ... unless someone knows of any usecase I'll just remove it ...

**Pull request merged**